MH4U Dex/MonsterDetailsViewController.m

```diff
@@ -29,39 +34,61 @@ @implementation MonsterDetailsViewController
 - (void)viewDidLoad
 {
     [super viewDidLoad];
-    [self.lowRankSubView setHidden:YES];
+    self.overviewSubView.hidden = NO;
+    self.lowRankSubView.hidden = YES;
+    self.highRankSubView.hidden = YES;
+    self.GRankSubView.hidden = YES;
 }
 
 - (IBAction)segmentedControlChanged:(UISegmentedControl *)segmentedControl
 {
     switch (segmentedControl.selectedSegmentIndex) {
         case Overview:
-            [self.overviewSubView setHidden:NO];
-            [self.lowRankSubView setHidden:YES];
+            self.overviewSubView.hidden = NO;
+            self.lowRankSubView.hidden = YES;
+            self.highRankSubView.hidden = YES;
+            self.GRankSubView.hidden = YES;
             self.navigationItem.title = [NSString stringWithFormat:@"%@ Overview", self.monster];
             break;
         case LowRank:
-            [self.overviewSubView setHidden:YES];
-            [self.lowRankSubView setHidden:NO];
+            self.overviewSubView.hidden = YES;
+            self.lowRankSubView.hidden = NO;
+            self.highRankSubView.hidden = YES;
+            self.GRankSubView.hidden = YES;
             self.navigationItem.title = [NSString stringWithFormat:@"%@ LR Drops", self.monster];
             break;
+        case HighRank:
+            self.overviewSubView.hidden = YES;
+            self.lowRankSubView.hidden = YES;
+            self.highRankSubView.hidden = NO;
+            self.GRankSubView.hidden = YES;
+        case GRank:
+            self.overviewSubView.hidden = YES;
+            self.lowRankSubView.hidden = YES;
+            self.highRankSubView.hidden = YES;
+            self.GRankSubView.hidden = NO;
         default:
             break;
     }
 }
 
-- (void)viewWillAppear:(BOOL)animated
-{
-    [super viewWillAppear:animated];
-    NSLog(@"Test");
-}
-
 - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
 {
     if ([segue.identifier isEqualToString:@"embedLowRank"]) {
-        MonsterLowRankTableViewController *lowRankVC = (MonsterLowRankTableViewController *)segue.destinationViewController;
+        MonsterItemTableViewController *lowRankVC = (MonsterItemTableViewController *)segue.destinationViewController;
         lowRankVC.managedObjectContext = self.managedObjectContext;
         lowRankVC.monster = self.monster;
+        lowRankVC.rank = @"Low";
+    } else if ([segue.identifier isEqualToString:@"embedHighRank"]) {
+        MonsterItemTableViewController *highRankVC = (MonsterItemTableViewController *)segue.destinationViewController;
+        highRankVC.managedObjectContext = self.managedObjectContext;
+        highRankVC.monster = self.monster;
+        highRankVC.rank = @"High";
+    } else if ([segue.identifier isEqualToString:@"embedGRank"]) {
+        MonsterItemTableViewController *GRankVC = (MonsterItemTableViewController *)segue.destinationViewController;
+        GRankVC.managedObjectContext = self.managedObjectContext;
+        GRankVC.monster = self.monster;
```

Comment:
These three lines are basically identical, except for the variable name, in each block of this `if`-`else`-`if` chain—you could probably move some form of it out to before the `if`s or otherwise restructure the `if`s to avoid the repetition.